### PR TITLE
Miscellaneous mini-"typos" fixes

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** fixes / typos

This is a collection of typographic/English fixes or specification improvements in vernac code, documentation, ... found over the last months.

Semantically, it also includes protecting two `try ... with _ -> ...`.

### CODE REVIEW

**Review by silene — APPROVED** [review]

**Review by jfehrle — APPROVED** [review]
LGTM

File: vernac/vernacentries.ml [hunk 1]
```diff
@@ -1937,7 +1937,7 @@ let vernac_search ~pstate ~atts s gopt r =
     match gopt with | None ->
       (* 1st goal by default if it exists, otherwise no goal at all *)
       (try get_goal_or_global_context ~pstate 1
-       with _ -> let env = Global.env () in (Evd.from_env env, env))
+       with e when CErrors.noncritical e -> let env = Global.env () in (Evd.from_env env, env))
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml:1940]: This `try` is actually dead code since 7efaf86882488034e6545505e1eda7d6e1a6ce14 started using `_or_global_context`

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
Just a small change needed

File: vernac/vernacentries.ml (hunk 1 above)
Comment (herbelin) [reply vernac/vernacentries.ml:1940]: So, you mean just keeping `get_goal_or_global_context ~pstate 1`? OK, then.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**